Kernel/Syscalls/unveil.cpp

```diff
@@ -35,7 +85,17 @@ ErrorOr<FlatPtr> Process::sys$unveil(Userspace<Syscall::SC_unveil_params const*>
         return 0;
     }
 
-    if (veil_state() == VeilState::Locked)
+    if (!((params.flags & static_cast<unsigned>(UnveilFlags::CurrentProgram)) || (params.flags & static_cast<unsigned>(UnveilFlags::AfterExec))))
```

Comment:
here perhaps a `to_underlying` is more appropriate than a static_cast. 

or we could make a local copy. `auto param_flags = static_cast<UnveilFlags>(params.flags)`. ... after checking bounds of the enum ofc. hm. maybe not, that sounds messy to do properly without undefined behavior.